src/c++/perf_analyzer/infer_data_manager_base.h

```diff
@@ -108,6 +108,11 @@ class InferDataManagerBase : public IInferDataManager {
 
   virtual cb::Error InitInferDataOutput(
       const std::string& name, InferData& infer_data) = 0;
+
+#ifndef DOCTEST_CONFIG_DISABLE
```

Comment:
I'm not saying I have a solution, but it feels really wrong that we need to add 4 default constructors hidden behind this define, as well as create multiple mocks and touch 6 test files just to test this small feature change.

I know part of the issue is just that when function or constructor prototypes change (in this case, LoadManager::InitManager) that a ton of files have to change.

I also think that lack of interfaces is causing us issues, and making us create these mock/real hybrid objects. If we had interfaces for all of these classes we could create clean mocks

Reply:
I agree that more interfaces would be cleaner code. This is just my way of allowing tests to create barebones objects so that any relevant data members can be set in the test, making it clear to the code reader what's important for that test. Also makes it easier in case creating inputs for the bigger constructors is hard.